smac/smbo/smbo.py

```diff
@@ -126,12 +127,16 @@ def run(self):
 
             self.logger.debug("Intensify")
 
-            self.incumbent, inc_perf = self.intensifier.intensify(
-                challengers=challengers,
-                incumbent=self.incumbent,
-                run_history=self.runhistory,
-                aggregate_func=self.aggregate_func,
-                time_bound=max(0.01, time_spend))
+            try: self.incumbent, inc_perf = self.intensifier.intensify(
+                  challengers=challengers,
+                  incumbent=self.incumbent,
+                  run_history=self.runhistory,
+                  aggregate_func=self.aggregate_func,
+                  time_bound=max(0.01, time_spend))
+            except (TAEAbortException, BudgetExhaustedException):
```

Comment:
Actually I suppose the `BudgetExhaustedException` does not need to be handed up to smbo, since there already is a check for budget there. Also changed the handling in intensify to be more intuitive.